### DISCUSSION TITLE
Push from action (even with PAT) does not trigger action

### DISCUSSION THREAD

**Discussion by gmalkov** [community/community#26220] [Actions]
<p>I have 2 workflows that you can see here:<br>
<a href="https://github.com/DataBiosphere/framework-env/tree/264a60fc614751b687e9d68e87bb225ba79f12d8/.github/workflows" rel="nofollow noopener">https://github.com/DataBiosphere/framework-env/tree/264a60fc614751b687e9d68e87bb225ba79f12d8/.github/workflows</a></p>
<p>One which reveives a dispatch and bumps a value in a JSON (update.yml), and another which should be triggered when the first commits the change.</p>
<p>As per this thread:</p>
<p><a href="https://github.community/t/github-action-not-being-triggered/16055">https://github.community/t5/GitHub-Actions/Github-Action-not-being-triggered/td-p/30111</a></p>
<p>I made sure to use a PAT for the push, but it still does not trigger the other action.</p>

**Comment by logankilpatrick** [comment]
<p>Hey! I am not clear what the designed workflow would be. Can you provide a high-level overview of how those two actions would interact? </p>

**Comment by gmalkov** [comment]
<p>The update action receives a dispatch containing some information based on which it updates a JSON in the repo and pushes it up.</p>
<p>I’d like that push to in turn trigger another action that performs further steps based on the changes made by the first. I’d like this JSON to also be modified by other actors than the first action, and still trigger the second action with their pushes.</p>

**Comment by airtower-luna** [comment]
<p>That is by design, accoding to the <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows" rel="nofollow noopener">documentation on events that trigger workflows</a>. Key quote:</p>
<p>“An action in a workflow run can’t trigger a new workflow run. For example, if an action pushes code using the repository’s GITHUB_TOKEN, a new workflow will not run even when the repository contains a workflow configured to run when push events occur.”</p>

**Comment by gmalkov** [comment]
<p>So you’re saying the approach of using a PAT as mentioned in the thread I linked is just a workaround contrary to the intended function, and it got patched?</p>
<p>If that’s the case I guess I can send a dispatch from the first action in addition to pushing up the code change, and make the second action trigger from either a dispatch or a push. Not sure I agree with actions not being able to launch other actions is a great pattern.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/gmalkov">@gmalkov</a> ,</p>
<p>‘actions/checkout@v2’ will persist the GITHUB_TOKEN credential to the repo config to make git operations like push…etc, which will not trigger the another workflow then.</p>
<p>Please use ‘actions/checkout@v1’ instead. Or if you use v2, please add ‘persist-credentials: false’ in the action args. </p>
<p>It will trigger new workflow with PAT then. Please refer to the <a href="https://github.community/t/action-does-not-trigger-another-on-push-tag-action/17148/4">link</a> for more details.</p>
<p>Thanks.</p>

**Comment by gmalkov** [comment]
<p>Aha, yes!</p>
<pre><code>- name: Checkout current code
uses: actions/checkout@v2
with:
persist-credentials: false
</code></pre>
<p>did the trick. Now a push with PAT works. Thanks!</p>

**Comment by sanjacob** [comment]
If you post a repo here, be sure not to later delete it, making this whole thread pointless 🙄
